src/dmtcp_coordinator.cpp

```diff
@@ -1282,18 +1283,29 @@ calcLocalAddr()
       if (error != 0) {
         JTRACE("getnameinfo() failed.") (gai_strerror(error));
         continue;
-      }
-      if (Util::strStartsWith(name, hostname) ||
-          Util::strStartsWith(hostname, name)) {
+      } else {
+        at_least_one_match = true;
```

Comment:
Why do we set this to true here? Shouldn't we be doing this in the if clause later?

```
if ( Util::strStartsWith(name, hostname) &&
     Util::strStartsWith(hostname, name) ) {
   ...
}
```